tests/timer/test_Event.cpp

```diff
@@ -0,0 +1,68 @@
+#include "gtest/gtest.h"
+
+#include "Kokkos_Core.hpp"
+
+#include "kokkos-utils/timer/Event.hpp"
+
+/**
+ * @file
+ *
+ * @addtogroup unittests
+ *
+ * Events in time
+ * --------------
+ *
+ * This group of tests check the behavior of the events in time defined in @ref Event.hpp.
+ */
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+
+namespace Kokkos::utils::tests::timer
+{
+
+using namespace Kokkos::utils::timer;
+
+using event_t = Event<execution_space>;
+
+//! @test Check that @ref event_t satisfies the concept @ref Kokkos::utils::timer::IsEvent.
+TEST(Event, is_event)
+{
+    static_assert(IsEvent<event_t, execution_space, microseconds>);
+    static_assert(IsEvent<event_t, execution_space, milliseconds>);
+    static_assert(IsEvent<event_t, execution_space, seconds>);
+}
+
+//! @test Check the type of the event used by the primary template and the device specializations.
+TEST(Event, impl_event_type)
+{
+    using expt_impl_event_t =
+#if defined(KOKKOS_ENABLE_CUDA)
+    cudaEvent_t;
+#elif defined(KOKKOS_ENABLE_HIP)
+    hipEvent_t;
+#elif defined(KOKKOS_ENABLE_OPENMP)
+    std::chrono::steady_clock::time_point;
+#else
+    #error "Default execution space must be Cuda, HIP or OpenMP."
+#endif
```

Comment:
Not sure why we restricted to `OpenMP`. I guess we just want the backends that are specialized.
```suggestion
#else
    std::chrono::steady_clock::time_point;
#endif
```